static/js/starred_messages.js

```diff
@@ -34,6 +34,27 @@ exports.get_starred_msg_ids = function () {
     return Array.from(exports.ids);
 };
 
+exports.unstar_topic = function (stream_id, topic) {
+    for (msg_id of exports.get_starred_msg_ids()) {
+        const message = message_store.get(msg_id);
+        if (message.stream_id === stream_id &&
+                message.topic.toLowerCase() === topic.toLowerCase()) {
+            message_flags.toggle_starred_and_update_server(message);
+        }
+    };
+};
+
+exports.is_topic_starred = function (stream_id, topic) {
+    for (msg_id of exports.get_starred_msg_ids()) {
+        const message = message_store.get(msg_id);
+        if (message.stream_id === stream_id &&
+                message.topic.toLowerCase() === topic.toLowerCase()) {
+            return true;
+        }
+    };
+    return false;
+}
```

Comment:
I think this logic is unsound, in that messages in  `starred_messages` aren't necessarily present in `message_store`.  

Maybe let's punt the "starred messages" commit for now?

Reply:
Let me try and see if we can fetch all the starred messages first. I believe folk have around 100-200 starred messages at max, shouldn't be too much.

Reply:
That's a rabbit hole that isn't worth adding as a dependency of this big change; just keep it on your TODO list for future ideas.